tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py

```diff
@@ -209,3 +209,52 @@ def test_plot_parallel_coordinate_unique_hyper_param() -> None:
     figure = plot_parallel_coordinate(study_categorical_params)
     assert len(figure.get_lines()) == 0
     plt.savefig(BytesIO())
+
+
+@pytest.mark.parametrize("value", [float("inf"), -float("inf"), float("nan")])
+def test_nonfinite_removed(value: float) -> None:
+
+    study = create_study()
+    study.add_trial(
+        create_trial(
+            value=0.0,
+            params={"param_a": 1e-6},
+            distributions={"param_a": FloatDistribution(0.0, 1.0)},
+        )
+    )
+    study.add_trial(
+        create_trial(
+            value=value,
+            params={"param_a": 0},
+            distributions={"param_a": FloatDistribution(0.0, 1.0)},
+        )
+    )
+
+    figure = plot_parallel_coordinate(study)
+    assert len(figure.get_lines()) == 0
```

Comment:
I think we are not testing correct thing here. This test will indeed fail on current master and pass with this PR,  but failure occurs before assertion due to `ValueError` raised by matplotlib functions, as described in #3382:
>optuna.visualization.matplotlib.plot_parallel_coordinate
>
>Current: ValueError: Axis limits cannot be NaN or Inf
>Expected: Plotted correctly

Following should be enough
```suggestion
    plot_parallel_coordinate(study)
```

Reply:
Thanks, got it.